boards/arm/nrf52840_pca10059/doc/nrf52840_pca10059.rst

```diff
@@ -0,0 +1,145 @@
+the slow clock is 32.768 kHz. The frequency of the main clock
+is 32 MHz.
+
+Supported Features
+==================
+
+The ``nrf52840_pca10059`` board configuration supports the following
+hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| NVIC      | on-chip    | nested vectored      |
+|           |            | interrupt controller |
++-----------+------------+----------------------+
+| RTC       | on-chip    | system clock         |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| FLASH     | on-chip    | flash                |
++-----------+------------+----------------------+
+| RADIO     | on-chip    | Bluetooth            |
++-----------+------------+----------------------+
+| RTT       | on-chip    | console              |
++-----------+------------+----------------------+
+
+Other hardware features are not supported by the Zephyr kernel.
+See `nRF52840 Dongle website`_ and `Nordic Semiconductor Infocenter`_
+for a complete list of nRF52840 PCA10059 Development Kit board hardware features.
+
+Connections and IOs
+===================
+
+LED
+---
+
+* LED1 (green) = P0.6
+* LED2 (red)   = P0.8
+* LED2 (green) = P1.9
+* LED2 (blue)  = P0.12
+
+Push buttons
+------------
+
+* BUTTON1 = SW1 = P1.6
+* RESET   = SW2 = P0.18
+
+Programming and Debugging
+*************************
+
+Applications for the ``nrf52840_pca10059`` board configuration can be
+built in the usual way (see :ref:`build_an_application` for more details);
+however, an external debugger IC is necessary to program the device,
+and the standard debugging targets are not currently available.
+
+Flashing
+========
+
+Flashing Zephyr onto the ``nrf52840_pca10059`` board requires an external
+J-Link programmer. The programmer is attached to the SWD header on the back
+side of the board.
+
+Follow the instructions in the :ref:`nordic_segger` page to install
+and configure all the necessary software. Further information can be
+found in :ref:`nordic_segger_flashing`. Then build and flash
+applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Here is an example for the :ref:`blinky` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nrf52840_pca10059
+   :goals: build flash
+
+Observe the LED on the board blinking.
+
+Debugging
+=========
+
+The ``nrf52840_pca10059`` board does not have an on-board J-Link debug IC
+as some nRF5x development boards, however, instructions from the
+:ref:`nordic_segger` page also apply to this board, with the additional step
+of connecting an external debugger.
+
+Testing the LEDs and buttons on the nRF52840 PCA10059
+************************************************
```

Comment:
title underline needs to be at least as long as the title, so add some more ``***``s